src/content/docs/plugin-development/recommended-workflow.mdx

```diff
@@ -0,0 +1,143 @@
+---
+title: Recommended Workflow
+description: Recommended Workflow
+sidebar:
+  order: 2
+---
+
+For an optimized and efficient development experience, we recommend adopting the following workflow for creating plugins for the Re:Earth Visualizer. This approach utilizes a plugin template designed with:
+
+- **React** as the front-end framework.
+- **ShadCN**, integrated with Tailwind CSS, as the UI solution.
+
+This workflow, rather than following the standard plugin installation process, enables the Re:Earth Visualizer to dynamically retrieve the plugin directly from the preview server. This approach significantly streamlines the development cycle by allowing faster and more efficient plugin testing and iteration.
+
+## Create New Plugin using Template
+
+For now we recommend using this <a href="https://github.com/reearth-plugins/reearth-visualizer-plugin-shadcn-template" target="_blank">Re:Earth Visualizer Plugin ShadCN Template</a>, we'll have more templates available in the future.
+
+Create your new plugin repository using the template.
+
+Navigate to the plugin directory and explore the structure of the template repository.
+
+```bash
+cd your-new-plugin
+```
+
+Install dependencies:
+
+```bash
+yarn install
+```
+
+Build plugin once for initialize the reqired files:
+
+```bash
+yarn build
+```
+
+Start devlepment & preview server:
+
+```bash
+yarn dev-build
+```
+
+:::note
+Now your plugin will be auto-build and serve on `http://localhost:5005`.
+:::
+
+:::tip
+For more information about the template please read its <a href="https://github.com/reearth-plugins/reearth-visualizer-plugin-shadcn-template" target="_blank">README.md</a>.
+:::
+
+## Develop plugin together with Visualizer
+
+import { LinkCard } from "@astrojs/starlight/components";
+
```

Comment:
Please follow this page to set up your visualizer application in the local environment.

Reply:
Updated